staging/src/k8s.io/apiextensions-apiserver/test/integration/ratcheting_test.go

```diff
@@ -0,0 +1,1396 @@
+					"field": map[string]interface{}{
+						"object1": map[string]interface{}{
+							"stringField": "a string",
+							"intField":    15,
+						},
+						"object2": map[string]interface{}{
+							"stringField":   "another string",
+							"intField":      10,
+							"otherIntField": 20,
+						},
+					},
+				}),
+				// Show can not change a grandfathered field to an invalid value
+				expectError(applyPatchOperation(myCRDV1Beta1, myCRDInstanceName, map[string]interface{}{
+					"field": map[string]interface{}{
+						"object2": map[string]interface{}{
+							"stringField": "a changed string",
+						},
+						"object3": map[string]interface{}{
+							"stringField": "a changed third string",
+						},
+					},
+				})),
+				// Add another restriction on the stringField
+				patchMyCRDV1Beta1Schema(map[string]interface{}{
+					"properties": map[string]interface{}{
+						"field": map[string]interface{}{
+							"additionalProperties": map[string]interface{}{
+								"stringField": map[string]interface{}{
+									"x-kubernetes-validations": []interface{}{
+										map[string]interface{}{
+											"rule":    "self.startsWith('k8s')",
+											"message": "strings must have k8s prefix",
+										},
+										map[string]interface{}{
+											"rule":    "len(self) % 2 == 1",
+											"message": "strings must have odd length",
+										},
+									},
+								},
+							},
+						},
+					},
+				}),
+				// Show we can have mixed ratcheting of one or two CEL rules
+				// object4 is ratcheted by one rule, object1 is ratcheting 2 rules
+				applyPatchOperation(myCRDV1Beta1, myCRDInstanceName, map[string]interface{}{
+					"field": map[string]interface{}{
+						"object1": map[string]interface{}{
+							"stringField": "a string", // invalid. even number length, no k8s prefix
+							"intField":    1000,
+						},
+						"object4": map[string]interface{}{
+							"stringField": "k8s third string", // invalid. even number length. ratcheted
+							"intField":    7000,
+						},
+					},
+				}),
+				// Show we get error with swapped keys
+				expectError(applyPatchOperation(myCRDV1Beta1, myCRDInstanceName, map[string]interface{}{
+					"field": map[string]interface{}{
+						"object1": map[string]interface{}{
+							"stringField": "k8s third string",
+							"intField":    1000,
+						},
+						"object4": map[string]interface{}{
+							"stringField": "a string",
+							"intField":    7000,
+						},
+					},
+				})),
+				// Show we can fix keys
+				expectError(applyPatchOperation(myCRDV1Beta1, myCRDInstanceName, map[string]interface{}{
+					"field": map[string]interface{}{
+						"object1": map[string]interface{}{
+							"stringField": "k8s a stringy",
+							"intField":    1000,
+						},
+						"object4": map[string]interface{}{
+							"stringField": "k8s third stringy",
+							"intField":    7000,
+						},
+					},
+				})),
+			},
+		},
+		// Blocked Functionality
```

Comment:
clarify what blocked means. Do these tests fail to ratchet?

Reply:
removed this comment, the names are already descriptive (and not all the tests are not filled in yet, deferring some of them to test freeze)